### ISSUE TITLE
"Referencing objects" Inspection shows a wrong message

### ISSUE THREAD

**Issue by ikeyan** [kalessil/phpinspectionsea#936]
The inspection says "Objects are always passed by reference; please correct '& $var'."
However, PHP function arguments are *passed by value* without `&`.
```PHP
function pass_by_val(DateTime $a) {
    $a = new DateTime("yesterday");
}
function pass_by_ref(DateTime &$a) {
    $a = new DateTime("yesterday");
}

$x = new DateTime();
pass_by_val($x);
var_dump($x); // 2018-06-08 19:36:23
pass_by_ref($x);
var_dump($x); // 2018-06-07 00:00:00
```

In `pass_by_val`, `$a` is passed by value, while the value of `$a` has a reference type.

If there is an inspection that warns a passed-by-ref argument (of any type) that isn't modified nor further passed by ref, that may be informative.
```PHP
function f(&$x) { // $x is not modified
  echo $x;
}
function g(&$x) { // OK
  var_dump([&$x]);
}

// false positive; though this must be a rare case
// $x is modified in $f
function while_loop(&$x, callable $f) { // $x is not modified
  while ($x) {
    $f();
  }
}
$x = 3;
while_loop($x, function () use (&$x) { echo $x--; }); // prints 321
```

**Comment by kalessil** [comment]
Thank you for reporting @ikeyan, I'll take care about the issue.

**Comment by kalessil** [comment]
Fixed!
